crates/services/block_aggregator_api/src/api/protobuf_adapter.rs

```diff
@@ -54,13 +68,13 @@ impl BlockAggregator for Server {
             ))),
         }
     }
-    type GetBlockRangeStream = ReceiverStream<Result<BlockResponse, Status>>;
+    type GetBlockRangeStream = ReceiverStream<Result<ProtoBlockResponse, Status>>;
```

Comment:
```suggestion
    type GetBlockRangeStream = BoxStream<Result<ProtoBlockResponse, Status>>;
```

If you used `BoxStream`, then you could avoid usage of the `tokio::spawn(` below. You can just do `inner.map({...}).into_boxed()`